mgmtsystem_action/models/mgmtsystem_action_report.py

```diff
@@ -0,0 +1,70 @@
+
+
+class MgmtsystemtActionReport(models.Model):
+    """Management System Action Report."""
+
+    _name = "mgmtsystem.action.report"
+    _auto = False
+    _description = "Management System Action Report"
+    _rec_name = 'id'
+
+    # Compute data
+    number_of_actions = fields.Integer('# of actions', readonly=True)
+    age = fields.Integer('Age', readonly=True)
+    number_of_days_to_open = fields.Integer('# of days to open', readonly=True)
+    number_of_days_to_close = fields.Integer(
+        '# of days to close', readonly=True)
+    number_of_exceedings_days = fields.Integer(
+        '# of exceedings days', readonly=True)
+
+    # Grouping view
+    type_action = fields.Selection([
+        ('immediate', 'Immediate Action'),
+        ('correction', 'Corrective Action'),
+        ('prevention', 'Preventive Action'),
+        ('improvement', 'Improvement Opportunity')
+    ], 'Response Type')
+    create_date = fields.Datetime('Create Date', readonly=True, select=True)
+    opening_date = fields.Datetime('Opening Date', readonly=True, select=True)
+    date_closed = fields.Datetime('Close Date', readonly=True, select=True)
+    date_deadline = fields.Date('Deadline', readonly=True, select=True)
+    user_id = fields.Many2one('res.users', 'User', readonly=True)
+    stage_id = fields.Many2one(
+        'mgmtsystem.action.stage', 'Stage', readonly=True)
+    system_id = fields.Many2one('mgmtsystem.system', 'System', readonly=True)
+
+    def init(self, cr):
+        """Display a pivot view of action."""
+        tools.drop_view_if_exists(cr, 'mgmtsystem_action_report')
+        cr.execute("""
+             CREATE OR REPLACE VIEW mgmtsystem_action_report AS (
+                 select
+                    m.id,
+                    m.date_closed as date_closed,
+                    m.date_deadline as date_deadline,
+                    m.opening_date as opening_date,
+                    m.user_id,
+                    m.stage_id,
+                    m.system_id,
+                    m.type_action as type_action,
+                    m.create_date as create_date,
+                    avg(extract('epoch' from (current_date-m.create_date))
+                    )/(3600*24) as  age,
+                    avg(extract('epoch' from (m.opening_date-m.create_date))
+                    )/(3600*24) as  number_of_days_to_open,
+                    avg(extract('epoch' from (m.date_closed-m.create_date))
+                    )/(3600*24) as  number_of_days_to_close,
+                    avg(extract('epoch' from (m.date_closed - m.date_deadline))
+                    )/(3600*24) as  number_of_exceedings_days,
+                    count(*) AS number_of_actions
+                from
+                    mgmtsystem_action m
+                group by m.user_id,m.system_id, m.stage_id, m.opening_date, \
+                        m.create_date,m.type_action,m.date_deadline, \
+                        m.date_closed, m.id
+            )
+            """)
```

Comment:
The problem with this report is that you cant drill down to grouped statistics into individual actions.
I propose that we implement the `number_of_days_to_X` computation directly in Actions, and just use the Pivot on these. The computation can be added in the already implemented `write()` method.
And with this, this report becomes obsolete.